### PR TITLE
[MF-301] Create toggle under user menu for changing language

### PR DESCRIPTION
**Pull request by samuelmale**
Demo:

Ticket: https://issues.openmrs.org/browse/MF-301

### CODE REVIEW

File: src/components/choose-locale/change-locale.component.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export const ChangeLocale: React.FC<ChangeLocaleProps> = ({
+  allowedLocales,
+  user
+}) => {
+  const [userProps, setUserProps] = useState(user.userProperties);
+  const options = allowedLocales?.map(locale => (
+    <option value={locale} key={locale}>
+      {locale}
+    </option>
+  ));
```
Comment (brandones) [inline src/components/choose-locale/change-locale.component.tsx:19]: My preference would be to inline this in the `return` block

File: src/components/choose-locale/change-locale.component.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  useEffect(() => {
+    if (user.userProperties.defaultLocale !== userProps.defaultLocale) {
+      updateUserProperties(user.uuid, userProps, new AbortController()).then(
+        response => {
+          if (response.ok) {
+            refetchCurrentUser();
+          }
+        }
+      );
```
Comment (brandones) [inline src/components/choose-locale/change-locale.component.tsx]: Return a cancelling function using the AbortController. See https://github.com/openmrs/openmrs-esm-login/blob/ba86336f68ddc4fa364534b605a4184fea5ac1eb/src/choose-location/choose-location.component.tsx#L71 for example.

File: src/components/choose-locale/change-locale.component.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    }
+  }, [userProps]);
+
+  return (
+    <div className="omrs-margin-12">
+      <label htmlFor="selectLocale">
+        <span>Select Locale</span>
```
Comment (brandones) [inline src/components/choose-locale/change-locale.component.tsx:36]: Let's not forget that we still need to internationalize primary-navigation.

File: src/components/choose-locale/choose-locale.component.test.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { render, fireEvent, wait } from "@testing-library/react";
+import { ChangeLocale } from "./change-locale.component";
+import * as backendController from "./change-locale.resource";
+
+describe(`<ChangeLocale />`, () => {
+  const setupLocaleSelect = async () => {
+    const wrapper = render(
```
Comment (brandones) [inline src/components/choose-locale/choose-locale.component.test.tsx]: Please read https://kentcdodds.com/blog/common-mistakes-with-react-testing-library .

In this particular case, the advice is: you don't need the return value from `render`; just use `screen`.

File: src/components/choose-locale/choose-locale.component.test.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        }}
+      />
+    );
+    await wait();
+
+    return wrapper.getByDisplayValue("en") as HTMLInputElement;
+  };
+
+  it("should have user's defaultLocale as initial value", async () => {
+    expect((await setupLocaleSelect()).value).toBe("en");
```
Comment (brandones) [inline src/components/choose-locale/choose-locale.component.test.tsx]: This test would be a little bit stronger if the defaultLocale was not `en` (e.g. `fr`)

File: src/components/choose-locale/change-locale.component.tsx (hunk 2 above)
Comment (brandones) [reply src/components/choose-locale/change-locale.component.tsx]: @samuelmale This is the only change I'm going to say is *strictly required* before I approve this.

File: src/components/choose-locale/change-locale.component.tsx (hunk 3 above)
Comment (samuelmale) [reply src/components/choose-locale/change-locale.component.tsx:36]: @brandones I thought for this specific label we should default to `en`. If I'm a user and If by mistake change locale to `fr` that I don't understand, I can understand my way back to revert this, if this label is not translated. 

See below(Reference Application):
screencapture-openmrs-spa-org-openmrs-adminui-myaccount-myAccount-page-2020-10-05-22_12_46

I thought the above is intentionally untranslated?

File: src/components/choose-locale/change-locale.component.tsx (hunk 2 above)
Comment (samuelmale) [reply src/components/choose-locale/change-locale.component.tsx]: Nice catch @brandones 

There are many other places in the microfrontends were we need to fix this. Most especially in the patient registration esm.

File: src/components/choose-locale/change-locale.component.tsx (hunk 3 above)
Comment (brandones) [reply src/components/choose-locale/change-locale.component.tsx:36]: Oh, interesting thought. Sure, let's leave it like that then.

File: src/components/choose-locale/change-locale.component.tsx (hunk 1 above)
Comment (samuelmale) [reply src/components/choose-locale/change-locale.component.tsx:19]: I had this inlined but it was automatically formatted by some js plugin at pre-commit. Maybe 'js lint'?

Comment (brandones) [reply src/components/choose-locale/change-locale.component.tsx:19]: Sorry Samuel, when I say "inlined" I don't mean "in a single line," but rather that you should just put this whole value directly where the value goes (in the return block below). No temporary variable assignment. It's not important though.

**Review by brandones — APPROVED** [review]
Nice work! Thanks!

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks :beers:!

**Pull request merged**